synapse/models/infotech.py

```diff
@@ -25,559 +24,770 @@ class SemverType(DataType):
+                ('it:exec:file:del', ('guid', {}), {
+                    'doc': 'An instance of a host deleting a file from a filesystem.',
+                }),
+                ('it:exec:file:read', ('guid', {}), {
+                    'doc': 'An instance of a host reading a file from a filesystem.',
+                }),
+                ('it:exec:file:write', ('guid', {}), {
+                    'doc': 'An instance of a host writing a file to a filesystem.',
+                }),
+                ('it:exec:reg:get', ('guid', {}), {
+                    'doc': 'An instance of a host getting a registry key.',
+                }),
+                ('it:exec:reg:set', ('guid', {}), {
+                    'doc': 'An instance of a host creating or setting a registry key',
+                }),
+                ('it:exec:reg:del', ('guid', {}), {
+                    'doc': 'An instance of a host deleting a registry key.',
+                })
             ),
-
             'forms': (
-
-                ('it:host', {}, [
-                    ('name', {'ptype': 'it:hostname', 'doc': 'The name of the host or system.'}),
-                    ('desc', {'ptype': 'str:txt', 'doc': 'A free-form description of the host.'}),
-
-                    # FIXME we probably eventually need a bunch of stuff here...
-                    ('ipv4', {'ptype': 'inet:ipv4', 'doc': 'The last known ipv4 address for the host.'}),
-                    ('latlong', {'ptype': 'geo:latlong', 'doc': 'The last known location for the node'}),
-                ]),
-
                 ('it:hostname', {}, ()),
-
-                ('it:sec:cve', {'ptype': 'it:sec:cve'}, [
-                    ('desc', {'ptype': 'str', 'doc': 'A free-form description of the CVE vulnerability.'}),
-                ]),
-
-                ('it:av:sig', {}, [
-                    ('soft', {'ptype': 'it:prod:soft', 'req': 1, 'ro': 1,
-                        'doc': 'The anti-virus product which contains the signature.'}),
-                    ('name', {'ptype': 'str:lwr', 'req': 1, 'ro': 1,
-                        'doc': 'The signature name.', 'ro': 1}),
-                    ('desc', {'ptype': 'str',
-                        'doc': 'A free-form description of the signature.'}),
-                    ('url', {'ptype': 'inet:url',
-                        'doc': 'A reference URL for information about the signature.'}),
-                ]),
-
-                ('it:av:filehit', {'ptype': 'it:av:filehit'}, [
-                    ('file', {'ptype': 'file:bytes',
-                        'doc': 'The file that triggered the signature hit.'}),
-                    ('sig', {'ptype': 'it:av:sig',
-                        'doc': 'The signature that the file triggered on.'}),
-                ]),
-
+                ('it:host', {}, (
+                    ('name', ('it:hostname', {}), {
+                        'doc': 'The name of the host or system.',
+                    }),
+                    ('desc', ('str', {}), {
```

Comment:
Do we need to do anything about `str:txt`s becoming `str`? I've been marking the ones I see